services/movie_session.py

```diff
@@ -0,0 +1,64 @@
+from datetime import datetime
+from typing import Any
+
+from django.core.exceptions import ObjectDoesNotExist
+from django.db.models import QuerySet
+
+from db.models import MovieSession
+
+
+def create_movie_session(
+        movie_show_time: datetime,
+        movie_id: int,
+        cinema_hall_id: int,
+) -> MovieSession:
+    return MovieSession.objects.create(
+        show_time=movie_show_time,
+        movie_id=movie_id,
+        cinema_hall_id=cinema_hall_id,
+    )
+
+
+def get_movies_sessions(session_date: str = "") -> QuerySet:
+    sessions = MovieSession.objects.all()
+    if session_date:
+        sessions = sessions.filter(
+            show_time__date=session_date
+        )
+    return sessions
+
+
+def get_movie_session_by_id(movie_id: int) -> MovieSession:
+    return MovieSession.objects.get(pk=movie_id)
+
+
+def update_movie_session(
+        session_id: int,
+        show_time: datetime | None = None,
+        movie_id: int | None = None,
+        cinema_hall_id: int | None = None
+) -> tuple[Any, Any]:
+    update_data = {}
+    if show_time:
+        update_data["show_time"] = show_time
+    if cinema_hall_id:
+        update_data["cinema_hall_id"] = cinema_hall_id
+    if movie_id:
+        update_data["movie_id"] = movie_id
+
+    if update_data:
+        try:

```

Comment:
just do 
```
if update_data:
        MovieSession.objects.filter(pk=session_id).update(**update_data)

Reply:
you don't need to try except here - if movie session doesn't exist with this pk - just let the error happen (in this scenario is acceptable)